src/Mod/TechDraw/Gui/QGIViewDimension.cpp

```diff
@@ -2656,6 +2656,16 @@ double QGIViewDimension::getDefaultIsoDimensionLineSpacing() const
     return Rez::appX(m_lineWidth * 2.0);
 }
 
+double QGIViewDimension::getIsoDimensionLineSpacing() const
+{
+    auto vp = static_cast<ViewProviderDimension*>(getViewProvider(getViewObject()));
+    double customLineSpacing = vp->LineSpacingISO.getValue();
+    if (customLineSpacing == -1.0) {  // By default -1.0 in gui until user defined
```

Comment:
Hardcoded by setting default gui value to -1.0. Any suggestions to how to better check if should use default or not? Using default needs to be an option if it needs to be relative to line width. Maybe a solution could be to set gui value to 2.0 and then returning `m_linewidth * customLineSpacing` in function?

Reply:
You made a new preference entry above, no?   The ViewProvider ctor would set the property for line spacing to the preference value using a getter from App/Preferences or Gui/PreferencesGui.   Same for initializing the UI - for create, take the value from preferences, for update, take the value from the ViewProvider.

Reply:
@WandererFan Not following along? If `-1.0`, it will use default instead.

Reply:
You don't decide about using the default here, you just use whatever value is in the ViewProvider's line spacing property.

The ViewProvider property is initialized to the default at creation.

Reply:
No. If the value in viewprovider is -1, it uses the default ISO value. And that value can only be calculated from linewidth and not known beforehand